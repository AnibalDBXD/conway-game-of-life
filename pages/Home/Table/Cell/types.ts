export interface ICellProps {
    isActive: boolean;
    onClick?: () => void;
}