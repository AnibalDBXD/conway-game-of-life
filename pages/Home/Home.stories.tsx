/* eslint-disable @typescript-eslint/no-empty-function */
import React from "react";
import { Story, Meta } from "@storybook/react";

import Home from ".";

export default {
  title: "Pages/Home",
  component: Home,
  parameters: {
    docs: {
      description: {
        component: "Home page"
      }
    },
  }
} as Meta;

const Template: Story = () => <Home />;

export const Primary = Template.bind({});