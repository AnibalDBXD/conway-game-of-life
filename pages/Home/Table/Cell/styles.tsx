import styled from "styled-components";
import { COLORS } from "../../../../common/styles";
import { ICellProps } from "./types";

export const CellSize = 25;

export const StyledCell = styled.td<ICellProps>`
    background-color: ${(props): string => props.isActive ? COLORS.WHITE : COLORS.BLACK};
    min-width: ${CellSize | 25}px;
    height: ${CellSize | 25}px;
    cursor: pointer;
`;