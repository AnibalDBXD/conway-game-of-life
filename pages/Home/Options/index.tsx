import React, { ChangeEvent, FormEvent, useRef, useEffect } from "react";
import { IOptions } from "./types";
import { StyledForm, StyledOptions, StyledTipsAndRulesContainer } from "./styles";
import PlayPause from "./PlayPause";

const Options = ({ setPause, isPause, setNumberOfColumnsAndRows, setTime }: IOptions): JSX.Element => {
  let newTime = null;
  const MAX_TIME = 10000;
  const MIN_TIME = 100;
  const DEFAULT_TIME = 200;

  let newColumnsAndRows = null;

  const timeRef = useRef<HTMLInputElement>(null);
  const columnsAndRowsRef = useRef<HTMLInputElement>(null);

  const onChangeTime = (event: ChangeEvent<HTMLInputElement>): void => {
    event.preventDefault();
    const currentTime = Number(event.target.value);
    if (currentTime > MAX_TIME) {
      timeRef.current.value = MAX_TIME + "";
      return;
    }
    if (currentTime < MIN_TIME) {
      newTime = MIN_TIME;
      return;
    }
    newTime = (currentTime);
  };

  const onChangeColumnsAndRows = (event: ChangeEvent<HTMLInputElement>): void => {
    event.preventDefault();
    newColumnsAndRows = ((Number(event.target.value)));
  };

  const handleSubmit = (event?: FormEvent<HTMLFormElement> | FormEvent<HTMLButtonElement>): void => {
    event?.preventDefault();
    if (newTime) {
      setTime(newTime);
      timeRef.current.value = "";
    }
    if (newColumnsAndRows) {
      setNumberOfColumnsAndRows(newColumnsAndRows);
      columnsAndRowsRef.current.value = "";
    }
  };

  const onPause = (event: KeyboardEvent): void => {
    if (event.key === " ") {
      setPause(!isPause);
    }
  };

  const handlePause = (): void => {
    setPause(!isPause);
    handleSubmit();
  };

  useEffect(() => {
    document.addEventListener("keydown", onPause);
    return (): void => document.removeEventListener("keydown", onPause);
  }, [isPause]);

  return (
    <StyledOptions>
      <StyledForm onSubmit={handleSubmit}>
        <div>
          <input type="number" ref={timeRef} placeholder="Time in miliseconds" readOnly={!isPause} onChange={onChangeTime} defaultValue={DEFAULT_TIME} min={MIN_TIME} max={MAX_TIME} />
        </div>
        <PlayPause pause={isPause} onClick={(): void => handlePause()} />
        <input
          ref={columnsAndRowsRef}
          placeholder="set number of columns and rows"
          disabled={!isPause}
          onChange={onChangeColumnsAndRows} />
      </StyledForm>
      <StyledTipsAndRulesContainer>
        <button>Tips</button>
        <button>Rules</button>
      </StyledTipsAndRulesContainer>
    </StyledOptions>
  );
};

export default Options;
