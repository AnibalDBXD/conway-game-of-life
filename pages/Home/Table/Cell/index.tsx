import { StyledCell } from "./styles";
import { ICellProps } from "./types";

const Cell: React.FC<ICellProps> = ({ isActive, onClick }): JSX.Element => {
  return (
    <StyledCell onClick={onClick} isActive={isActive} />
  );
};

export default Cell;
