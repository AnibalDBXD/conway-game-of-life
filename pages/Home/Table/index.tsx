import React, { useState } from 'react';
import { StyledTableContainer } from './styles';
import { ICoordinates } from './types';
import { SearchActiveCellPredicate, Columns } from "./utils";

const Table = (): JSX.Element => {
  const [ActiveCells, setActiveCells] = useState <ICoordinates[]>([]);

  const SearchActiveCell = ({ X: currentX, Y: currentY }: ICoordinates): boolean =>
    ActiveCells.some(({ X: activeX, Y: activeY }) =>
      SearchActiveCellPredicate({ activeX, activeY, currentY, currentX }));

  const handleCellClick = ({ X, Y }: ICoordinates): void => {
    if (SearchActiveCell({ X, Y })) {
      const currentActiveCells = [...ActiveCells];
      const ActiveCellIndex = currentActiveCells.findIndex(({ X: activeX, Y: activeY }) =>
        SearchActiveCellPredicate({ activeX, activeY, currentY: Y, currentX: X }));
      currentActiveCells.splice(ActiveCellIndex, 1);
      setActiveCells(currentActiveCells);
    } else {
      setActiveCells([...ActiveCells, {X, Y}]);
    }
  };

  return (
    <StyledTableContainer hideScrollbars={false} component="main">
      <table>
        <tbody>
          {Columns.map((Colunm, columnI) => (
            <tr key={columnI}>
              {Colunm.map(({Component: Row}, rowI: number) => (
                <Row isActive={SearchActiveCell({ X: rowI, Y: columnI})} onClick={(): void => handleCellClick({ X: rowI, Y: columnI})} key={rowI} />
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </StyledTableContainer>
  );
};

export default Table;
