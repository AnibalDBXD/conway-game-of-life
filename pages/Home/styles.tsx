import styled from "styled-components";
import { COLORS } from "../../common/styles";

export const StyledHome = styled.div`
    background-color: ${COLORS.NIGHT_RIDER};
`;