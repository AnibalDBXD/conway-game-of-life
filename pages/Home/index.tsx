import { useState } from "react";
import Table from "./Table";
import Options from "./Options";
import { StyledHome } from "./styles";

const Home = (): JSX.Element => {
  const [numberOfColumnsAndRows, setNumberOfColumnsAndRows] = useState(10);
  const [time, setTime] = useState(200);
  const [isPause, setPause] = useState(true);

  return (
    <StyledHome>
      <Table isPause={isPause} numberOfColumnsAndRows={numberOfColumnsAndRows} time={time} />
      <Options
        setTime={(time: number): void => setTime(time)}
        setNumberOfColumnsAndRows={(number: number): void => setNumberOfColumnsAndRows(number)}
        isPause={isPause}
        setPause={(value: boolean): void => setPause(value)} />
    </StyledHome>
  );
};

export default Home;
