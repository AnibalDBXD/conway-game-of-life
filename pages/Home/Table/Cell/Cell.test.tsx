import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Primary } from './Cell.stories';
import "jest-styled-components";
import { CellDeath, CellLive } from '../../Home.test';

describe('<Cell />', () => {
  test('Render', () => {
    const { asFragment } = render(<Primary />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Kill and revive cell', () => {
    const { getByTestId } = render(<Primary />);
    const cell00 = getByTestId("00");
    fireEvent.click(cell00);
    expect(cell00).toHaveStyle(CellLive);
    fireEvent.click(cell00);
    expect(cell00).toHaveStyle(CellDeath);
  });
});