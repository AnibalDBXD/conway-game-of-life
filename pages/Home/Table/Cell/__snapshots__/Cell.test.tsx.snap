// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> Render 1`] = `<DocumentFragment />`;

exports[`<Home /> Render 1`] = `
<DocumentFragment>
  .c3 {
  background-color: #252525;
  min-width: 25px;
  height: 25px;
  cursor: pointer;
}

.c1 {
  background-color: #313131;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 80vh;
}

.c2 {
  box-shadow: 1px 4px 30px -9px rgb(0 0 0 / 66%);
  background-color: #FFFFFF;
}

.c4 {
  background: linear-gradient(180deg,#313131 7%,#252525 19%);
  height: 20vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c10 {
  border: 0;
  background: transparent;
  box-sizing: border-box;
  width: 0;
  height: 74px;
  border-color: transparent transparent transparent #FFFFFF;
  -webkit-transition: 100ms all ease;
  transition: 100ms all ease;
  cursor: pointer;
  border-style: solid;
  border-width: 37px 0 37px 60px;
}

.c10:hover {
  border-color: transparent transparent transparent #8E8E8E;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c8 {
  color: #FFFFFF;
  font-size: 16px;
  font-weight: bold;
}

.c9 {
  font-size: 24px;
  outline: 0;
  border: 0;
  width: 150px;
  border-radius: 10px;
  padding-left: 10px;
  box-shadow: -1px 10px 41px 1px rgba(0,0,0,0.64);
}

.c12 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  background-color: transparent;
  border: 2px solid #FFFFFF;
  border-radius: 8px;
  color: #FFFFFF;
  font-weight: bold;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.1s ease;
  -webkit-transition: transform 0.1s ease;
  transition: transform 0.1s ease;
}

.c12:hover {
  border-color: #FFFFFFAE;
  color: #FFFFFFAE;
}

.c12:active {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.c0 {
  background-color: #313131;
}

<div
    class="c0"
  >
    <main
      class="c1 indiana-scroll-container indiana-scroll-container--hide-scrollbars"
    >
      <table
        class="c2"
      >
        <tbody>
          <tr>
            <td
              class="c3"
              data-testid="00"
            />
            <td
              class="c3"
              data-testid="01"
            />
            <td
              class="c3"
              data-testid="02"
            />
            <td
              class="c3"
              data-testid="03"
            />
            <td
              class="c3"
              data-testid="04"
            />
            <td
              class="c3"
              data-testid="05"
            />
            <td
              class="c3"
              data-testid="06"
            />
            <td
              class="c3"
              data-testid="07"
            />
            <td
              class="c3"
              data-testid="08"
            />
            <td
              class="c3"
              data-testid="09"
            />
            <td
              class="c3"
              data-testid="010"
            />
            <td
              class="c3"
              data-testid="011"
            />
            <td
              class="c3"
              data-testid="012"
            />
            <td
              class="c3"
              data-testid="013"
            />
            <td
              class="c3"
              data-testid="014"
            />
            <td
              class="c3"
              data-testid="015"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="10"
            />
            <td
              class="c3"
              data-testid="11"
            />
            <td
              class="c3"
              data-testid="12"
            />
            <td
              class="c3"
              data-testid="13"
            />
            <td
              class="c3"
              data-testid="14"
            />
            <td
              class="c3"
              data-testid="15"
            />
            <td
              class="c3"
              data-testid="16"
            />
            <td
              class="c3"
              data-testid="17"
            />
            <td
              class="c3"
              data-testid="18"
            />
            <td
              class="c3"
              data-testid="19"
            />
            <td
              class="c3"
              data-testid="110"
            />
            <td
              class="c3"
              data-testid="111"
            />
            <td
              class="c3"
              data-testid="112"
            />
            <td
              class="c3"
              data-testid="113"
            />
            <td
              class="c3"
              data-testid="114"
            />
            <td
              class="c3"
              data-testid="115"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="20"
            />
            <td
              class="c3"
              data-testid="21"
            />
            <td
              class="c3"
              data-testid="22"
            />
            <td
              class="c3"
              data-testid="23"
            />
            <td
              class="c3"
              data-testid="24"
            />
            <td
              class="c3"
              data-testid="25"
            />
            <td
              class="c3"
              data-testid="26"
            />
            <td
              class="c3"
              data-testid="27"
            />
            <td
              class="c3"
              data-testid="28"
            />
            <td
              class="c3"
              data-testid="29"
            />
            <td
              class="c3"
              data-testid="210"
            />
            <td
              class="c3"
              data-testid="211"
            />
            <td
              class="c3"
              data-testid="212"
            />
            <td
              class="c3"
              data-testid="213"
            />
            <td
              class="c3"
              data-testid="214"
            />
            <td
              class="c3"
              data-testid="215"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="30"
            />
            <td
              class="c3"
              data-testid="31"
            />
            <td
              class="c3"
              data-testid="32"
            />
            <td
              class="c3"
              data-testid="33"
            />
            <td
              class="c3"
              data-testid="34"
            />
            <td
              class="c3"
              data-testid="35"
            />
            <td
              class="c3"
              data-testid="36"
            />
            <td
              class="c3"
              data-testid="37"
            />
            <td
              class="c3"
              data-testid="38"
            />
            <td
              class="c3"
              data-testid="39"
            />
            <td
              class="c3"
              data-testid="310"
            />
            <td
              class="c3"
              data-testid="311"
            />
            <td
              class="c3"
              data-testid="312"
            />
            <td
              class="c3"
              data-testid="313"
            />
            <td
              class="c3"
              data-testid="314"
            />
            <td
              class="c3"
              data-testid="315"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="40"
            />
            <td
              class="c3"
              data-testid="41"
            />
            <td
              class="c3"
              data-testid="42"
            />
            <td
              class="c3"
              data-testid="43"
            />
            <td
              class="c3"
              data-testid="44"
            />
            <td
              class="c3"
              data-testid="45"
            />
            <td
              class="c3"
              data-testid="46"
            />
            <td
              class="c3"
              data-testid="47"
            />
            <td
              class="c3"
              data-testid="48"
            />
            <td
              class="c3"
              data-testid="49"
            />
            <td
              class="c3"
              data-testid="410"
            />
            <td
              class="c3"
              data-testid="411"
            />
            <td
              class="c3"
              data-testid="412"
            />
            <td
              class="c3"
              data-testid="413"
            />
            <td
              class="c3"
              data-testid="414"
            />
            <td
              class="c3"
              data-testid="415"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="50"
            />
            <td
              class="c3"
              data-testid="51"
            />
            <td
              class="c3"
              data-testid="52"
            />
            <td
              class="c3"
              data-testid="53"
            />
            <td
              class="c3"
              data-testid="54"
            />
            <td
              class="c3"
              data-testid="55"
            />
            <td
              class="c3"
              data-testid="56"
            />
            <td
              class="c3"
              data-testid="57"
            />
            <td
              class="c3"
              data-testid="58"
            />
            <td
              class="c3"
              data-testid="59"
            />
            <td
              class="c3"
              data-testid="510"
            />
            <td
              class="c3"
              data-testid="511"
            />
            <td
              class="c3"
              data-testid="512"
            />
            <td
              class="c3"
              data-testid="513"
            />
            <td
              class="c3"
              data-testid="514"
            />
            <td
              class="c3"
              data-testid="515"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="60"
            />
            <td
              class="c3"
              data-testid="61"
            />
            <td
              class="c3"
              data-testid="62"
            />
            <td
              class="c3"
              data-testid="63"
            />
            <td
              class="c3"
              data-testid="64"
            />
            <td
              class="c3"
              data-testid="65"
            />
            <td
              class="c3"
              data-testid="66"
            />
            <td
              class="c3"
              data-testid="67"
            />
            <td
              class="c3"
              data-testid="68"
            />
            <td
              class="c3"
              data-testid="69"
            />
            <td
              class="c3"
              data-testid="610"
            />
            <td
              class="c3"
              data-testid="611"
            />
            <td
              class="c3"
              data-testid="612"
            />
            <td
              class="c3"
              data-testid="613"
            />
            <td
              class="c3"
              data-testid="614"
            />
            <td
              class="c3"
              data-testid="615"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="70"
            />
            <td
              class="c3"
              data-testid="71"
            />
            <td
              class="c3"
              data-testid="72"
            />
            <td
              class="c3"
              data-testid="73"
            />
            <td
              class="c3"
              data-testid="74"
            />
            <td
              class="c3"
              data-testid="75"
            />
            <td
              class="c3"
              data-testid="76"
            />
            <td
              class="c3"
              data-testid="77"
            />
            <td
              class="c3"
              data-testid="78"
            />
            <td
              class="c3"
              data-testid="79"
            />
            <td
              class="c3"
              data-testid="710"
            />
            <td
              class="c3"
              data-testid="711"
            />
            <td
              class="c3"
              data-testid="712"
            />
            <td
              class="c3"
              data-testid="713"
            />
            <td
              class="c3"
              data-testid="714"
            />
            <td
              class="c3"
              data-testid="715"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="80"
            />
            <td
              class="c3"
              data-testid="81"
            />
            <td
              class="c3"
              data-testid="82"
            />
            <td
              class="c3"
              data-testid="83"
            />
            <td
              class="c3"
              data-testid="84"
            />
            <td
              class="c3"
              data-testid="85"
            />
            <td
              class="c3"
              data-testid="86"
            />
            <td
              class="c3"
              data-testid="87"
            />
            <td
              class="c3"
              data-testid="88"
            />
            <td
              class="c3"
              data-testid="89"
            />
            <td
              class="c3"
              data-testid="810"
            />
            <td
              class="c3"
              data-testid="811"
            />
            <td
              class="c3"
              data-testid="812"
            />
            <td
              class="c3"
              data-testid="813"
            />
            <td
              class="c3"
              data-testid="814"
            />
            <td
              class="c3"
              data-testid="815"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="90"
            />
            <td
              class="c3"
              data-testid="91"
            />
            <td
              class="c3"
              data-testid="92"
            />
            <td
              class="c3"
              data-testid="93"
            />
            <td
              class="c3"
              data-testid="94"
            />
            <td
              class="c3"
              data-testid="95"
            />
            <td
              class="c3"
              data-testid="96"
            />
            <td
              class="c3"
              data-testid="97"
            />
            <td
              class="c3"
              data-testid="98"
            />
            <td
              class="c3"
              data-testid="99"
            />
            <td
              class="c3"
              data-testid="910"
            />
            <td
              class="c3"
              data-testid="911"
            />
            <td
              class="c3"
              data-testid="912"
            />
            <td
              class="c3"
              data-testid="913"
            />
            <td
              class="c3"
              data-testid="914"
            />
            <td
              class="c3"
              data-testid="915"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="100"
            />
            <td
              class="c3"
              data-testid="101"
            />
            <td
              class="c3"
              data-testid="102"
            />
            <td
              class="c3"
              data-testid="103"
            />
            <td
              class="c3"
              data-testid="104"
            />
            <td
              class="c3"
              data-testid="105"
            />
            <td
              class="c3"
              data-testid="106"
            />
            <td
              class="c3"
              data-testid="107"
            />
            <td
              class="c3"
              data-testid="108"
            />
            <td
              class="c3"
              data-testid="109"
            />
            <td
              class="c3"
              data-testid="1010"
            />
            <td
              class="c3"
              data-testid="1011"
            />
            <td
              class="c3"
              data-testid="1012"
            />
            <td
              class="c3"
              data-testid="1013"
            />
            <td
              class="c3"
              data-testid="1014"
            />
            <td
              class="c3"
              data-testid="1015"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="110"
            />
            <td
              class="c3"
              data-testid="111"
            />
            <td
              class="c3"
              data-testid="112"
            />
            <td
              class="c3"
              data-testid="113"
            />
            <td
              class="c3"
              data-testid="114"
            />
            <td
              class="c3"
              data-testid="115"
            />
            <td
              class="c3"
              data-testid="116"
            />
            <td
              class="c3"
              data-testid="117"
            />
            <td
              class="c3"
              data-testid="118"
            />
            <td
              class="c3"
              data-testid="119"
            />
            <td
              class="c3"
              data-testid="1110"
            />
            <td
              class="c3"
              data-testid="1111"
            />
            <td
              class="c3"
              data-testid="1112"
            />
            <td
              class="c3"
              data-testid="1113"
            />
            <td
              class="c3"
              data-testid="1114"
            />
            <td
              class="c3"
              data-testid="1115"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="120"
            />
            <td
              class="c3"
              data-testid="121"
            />
            <td
              class="c3"
              data-testid="122"
            />
            <td
              class="c3"
              data-testid="123"
            />
            <td
              class="c3"
              data-testid="124"
            />
            <td
              class="c3"
              data-testid="125"
            />
            <td
              class="c3"
              data-testid="126"
            />
            <td
              class="c3"
              data-testid="127"
            />
            <td
              class="c3"
              data-testid="128"
            />
            <td
              class="c3"
              data-testid="129"
            />
            <td
              class="c3"
              data-testid="1210"
            />
            <td
              class="c3"
              data-testid="1211"
            />
            <td
              class="c3"
              data-testid="1212"
            />
            <td
              class="c3"
              data-testid="1213"
            />
            <td
              class="c3"
              data-testid="1214"
            />
            <td
              class="c3"
              data-testid="1215"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="130"
            />
            <td
              class="c3"
              data-testid="131"
            />
            <td
              class="c3"
              data-testid="132"
            />
            <td
              class="c3"
              data-testid="133"
            />
            <td
              class="c3"
              data-testid="134"
            />
            <td
              class="c3"
              data-testid="135"
            />
            <td
              class="c3"
              data-testid="136"
            />
            <td
              class="c3"
              data-testid="137"
            />
            <td
              class="c3"
              data-testid="138"
            />
            <td
              class="c3"
              data-testid="139"
            />
            <td
              class="c3"
              data-testid="1310"
            />
            <td
              class="c3"
              data-testid="1311"
            />
            <td
              class="c3"
              data-testid="1312"
            />
            <td
              class="c3"
              data-testid="1313"
            />
            <td
              class="c3"
              data-testid="1314"
            />
            <td
              class="c3"
              data-testid="1315"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="140"
            />
            <td
              class="c3"
              data-testid="141"
            />
            <td
              class="c3"
              data-testid="142"
            />
            <td
              class="c3"
              data-testid="143"
            />
            <td
              class="c3"
              data-testid="144"
            />
            <td
              class="c3"
              data-testid="145"
            />
            <td
              class="c3"
              data-testid="146"
            />
            <td
              class="c3"
              data-testid="147"
            />
            <td
              class="c3"
              data-testid="148"
            />
            <td
              class="c3"
              data-testid="149"
            />
            <td
              class="c3"
              data-testid="1410"
            />
            <td
              class="c3"
              data-testid="1411"
            />
            <td
              class="c3"
              data-testid="1412"
            />
            <td
              class="c3"
              data-testid="1413"
            />
            <td
              class="c3"
              data-testid="1414"
            />
            <td
              class="c3"
              data-testid="1415"
            />
          </tr>
          <tr>
            <td
              class="c3"
              data-testid="150"
            />
            <td
              class="c3"
              data-testid="151"
            />
            <td
              class="c3"
              data-testid="152"
            />
            <td
              class="c3"
              data-testid="153"
            />
            <td
              class="c3"
              data-testid="154"
            />
            <td
              class="c3"
              data-testid="155"
            />
            <td
              class="c3"
              data-testid="156"
            />
            <td
              class="c3"
              data-testid="157"
            />
            <td
              class="c3"
              data-testid="158"
            />
            <td
              class="c3"
              data-testid="159"
            />
            <td
              class="c3"
              data-testid="1510"
            />
            <td
              class="c3"
              data-testid="1511"
            />
            <td
              class="c3"
              data-testid="1512"
            />
            <td
              class="c3"
              data-testid="1513"
            />
            <td
              class="c3"
              data-testid="1514"
            />
            <td
              class="c3"
              data-testid="1515"
            />
          </tr>
        </tbody>
      </table>
    </main>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <label
              class="c8"
              for="Time (miliseconds)"
            >
              Time (miliseconds)
            </label>
            <input
              class="c9"
              id="Time (miliseconds)"
              max="1000"
              min="100"
              type="range"
              value="200"
            />
          </div>
        </div>
        <button
          class="c10"
          data-testid="PlayPauseButton"
        />
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <label
              class="c8"
              for="Columns and Rows"
            >
              Columns and Rows
            </label>
            <input
              class="c9"
              id="Columns and Rows"
              max="40"
              min="8"
              type="range"
              value="16"
            />
          </div>
        </div>
      </div>
      <div
        class="c11"
      >
        <button
          aria-describedby="popup-1"
          class="c12"
        >
          Tips
        </button>
        <button
          class="c12"
        >
          Apply changes
        </button>
        <button
          class="c12"
        >
          Reset
        </button>
        <button
          aria-describedby="popup-2"
          class="c12"
        >
          Rules
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
